include/rive/playable.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef _RIVE_PLAYABLE_HPP_
+#define _RIVE_PLAYABLE_HPP_
+
+#include "rive/animation/loop.hpp"
+#include "rive/math/aabb.hpp"
+#include "rive/math/vec2d.hpp"
+#include <string>
+
+namespace rive {
+    class ArtboardInstance;
+    class Renderer;
+
+    class SMIInput;
+    class SMIBool;
+    class SMINumber;
+    class SMITrigger;
+
+    class Playable {    
+    protected:
+        ArtboardInstance* m_ArtboardInstance;
+
+        Playable(ArtboardInstance*);
+
+    public:
+        virtual ~Playable() {}
+    
+        float width() const;
+        float height() const;
+        AABB bounds() const { return {0, 0, this->width(), this->height()}; }
+
+        virtual std::string name() const = 0;
+
+        virtual Loop loop() const = 0;
+        // Returns true iff the playable is known to not be fully opaque
+        virtual bool isTranslucent() const = 0;
+        // returns -1 for continuous
+        virtual float durationSeconds() const = 0;
+    
+        // If applicaable, resets the internal time to the start of the animation
+        virtual void restart() = 0;
+        // returns true if draw() should be called
+        virtual bool advanceAndApply(float elapsedSeconds) = 0;
+
+        void draw(Renderer*);
+
+        virtual void pointerDown(Vec2D);
+        virtual void pointerMove(Vec2D);
+        virtual void pointerUp(Vec2D);
+
+        virtual size_t inputCount() const;
+        virtual SMIInput* input(size_t index) const;
```

Comment:
Wonder if we should later rename these to PlayableInput. 

Regarding naming, @neurowave what does the rest of the team think about the name Playable? It's a little chewy, but I can't think of something shorter and as meaningful (Scene? Clip? Show?). In case it's not clear, this is basically the entry point from the client's/embedder's perspective. It abstracts whether an animation, state machine, combination of, or whatever future system is actually making the changes to the hierarchy (and potentially even multiple hierarchies). I think Scene is probably the best metaphor that matches some of our other ones.

So the Playable becomes the Scene, the inputs become the SceneInput (SceneBool, SceneNumber, SceneTrigger) eventually we can have SceneEvents/Cues. Kind of fits?